### PR TITLE
MigrateMessageCode to manage new case sensitive md5 hash

### PR DESCRIPTION
**Pull request by matteotrubini**
Ref: https://github.com/wintercms/wn-translate-plugin/pull/6

### CODE REVIEW

File: updates/v2.1.0/migrate_message_code.php [hunk 1]
```diff
@@ -11,9 +12,28 @@ class MigrateMessageCode extends Migration
...
+            Schema::table(self::TABLE_NAME, function($table) {
+                $table->string('code_pre_2_1_0')->index()->nullable();
+            });
+        }
+
         foreach (Message::all() as $message) {
-            $default_message = $message->message_data['x'];
-            $message->code = Message::makeMessageCode($default_message);
+            $message->code_pre_2_1_0 = $message->code;
+            $message->code = $message->code_md5 ?: Message::makeMessageCode($message->message_data[Message::DEFAULT_LOCALE]);
```
Comment (mjauvin) [inline updates/v2.1.0/migrate_message_code.php:23]: Where does this `$message->code_md5` property comes from @matteotrubini ?

File: updates/v2.1.0/migrate_message_code.php [hunk 2]
```diff
@@ -11,9 +12,28 @@ class MigrateMessageCode extends Migration
...
+            $message->save();
+        }
+
+        if (Schema::hasColumn(self::TABLE_NAME, 'code_md5')) {
```
Comment (mjauvin) [inline updates/v2.1.0/migrate_message_code.php:27]: Same thing here.

File: updates/v2.1.0/migrate_message_code.php (hunk 1 above)
Comment (matteotrubini) [reply updates/v2.1.0/migrate_message_code.php:23]: `code_md5` col is created on `down()` and populated from `code` before it's recalculated with the old method.
This follow the same logic of `code_pre_2_1_0`: preserve old values on migrations to permit checks and debug.

File: updates/v2.1.0/migrate_message_code.php (hunk 2 above)
Comment (matteotrubini) [reply updates/v2.1.0/migrate_message_code.php:27]: See above

**Comment by mjauvin** [comment]
Thanks @matteotrubini

**Pull request merged**